src/app/project/description/description.component.ts

```diff
@@ -0,0 +1,276 @@
+     */
+    onValueChanged(data?: any) {
+
+        if (!this.form) {
+            return;
+        }
+
+        const form = this.form;
+
+        Object.keys(this.formErrors).map(field => {
+            this.formErrors[field] = '';
+            const control = form.get(field);
+            if (control && control.dirty && !control.valid) {
+                const messages = this.validationMessages[field];
+                Object.keys(control.errors).map(key => {
+                    this.formErrors[field] += messages[key] + ' ';
+                });
+
+            }
+        });
+    }
+
+    getProject() {
+        // get the project data from cache
+        this._cache.get(this.projectUuid).subscribe(
+            (response: ReadProject) => {
+                this.project = response;
+
+                // is logged-in user projectAdmin?
+                if (this._session.getSession()) {
+                    this.projectAdmin = this.sysAdmin ? this.sysAdmin : this.session.user.projectAdmin.some(e => e === this.project.id);
+                }
+
+                this.buildForm();
+                this.loading = false;
+            },
+            (error: ApiResponseError) => {
+                this._errorHandler.showMessage(error);
+            }
+        );
+
+        this.loading = false;
+    }
+
+    buildForm(): void {
+        // create deep copy of description array because it contains objects
+        this.description = JSON.parse(JSON.stringify(this.project.description));
+
+        // create shallow copy of keywords array
+        this.keywords = this.project.keywords.slice();
+
+        this.form = this._fb.group({
+            'longname': new FormControl({
+                value: this.project.longname, disabled: false
+            }, [ Validators.required ]),
+            'keywords': new FormControl({
+                // must be empty (even in edit mode), because of the mat-chip-list
+                value: [], disabled: false
+            })
+        });
+
+        this.form.valueChanges
+            .subscribe(data => this.onValueChanged(data));
+    }
+
+    /**
+     * gets string literal
+     * @param data
+     */
+    getStringLiteral(data: StringLiteral[]) {
+        this.description = data;
+        if (!this.description.length) {
+            this.formErrors['description'] = this.validationMessages['description'].required;
+        } else {
+            this.formErrors['description'] = '';
+        }
+    }
+
+    addKeyword(event: MatChipInputEvent): void {
+        const input = event.chipInput.inputElement;
+        const value = event.value;
+
+        if (!this.keywords) {
+            this.keywords = [];
+        }
+
+        // add keyword
+        if ((value || '').trim()) {
+            this.keywords.push(value.trim());
+        }
+
+        // reset the input value
+        if (input) {
+            input.value = '';
+        }
+    }
+
+    removeKeyword(keyword: any): void {
```

Comment:
_any_ should be replaced with _string_

Reply:
done in [ab03e17](https://github.com/dasch-swiss/dsp-app/pull/928/commits/ab03e177a0cf0cce96f985c7abb728fd33929a5c)